server/asset/eth/eth.go

```diff
@@ -332,9 +336,13 @@ func (eth *Backend) ValidateCoinID(coinID []byte) (string, error) {
 }
 
 // ValidateContract ensures that the swap contract is constructed properly, and
-// contains valid sender and receiver addresses.
-func (eth *Backend) ValidateContract(contract []byte) error {
-	return notImplementedErr
+// contains valid counterparty, secret hash, and locktime.
+func (eth *Backend) ValidateContract(txdata []byte) error {
+	_, _, _, err := dexeth.ParseInitiateData(txdata)
+	if err != nil {
+		return fmt.Errorf("unable to parse contract txdata: %w", err)
+	}
+	return nil
 }
```

Comment:
I think this is the essential part of this PR after the changes re #1154 / #1243.